appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/NonPersistentEJBTimerService.java

```diff
@@ -106,17 +127,20 @@ protected boolean cancelNonPersistentTimer(TimerPrimaryKey timerId)
             throws FinderException, Exception {
 
         RuntimeTimerState rt = getNonPersistentTimerState(timerId);
-        if (rt != null) {
-            if (rt.isCancelled()) {
-                // already cancelled or removed
-                return true;
-            }
-
-            cancelTimerSynchronization(null, timerId,
-                    rt.getContainerId(), ownerIdOfThisServer_, false);
+        if (rt == null) {
+            return false;
```

Comment:
In case of cancellation, If a timer is null or already canceled then the container must throw `javax.ejb.NoSuchObjectLocalException` instead of `FinderException` (EJB Spec 3.2 - Section 13.4.4 Timer Cancellation).

Reply:
Yes. This exception is wrapped into `NoSuchObjectLocalException` in `TimerWrapper`